app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    stored_cache = {}
+
+    def wrapper(*args) -> Any:
+        if args in stored_cache:
+            print("Getting from cache")
+            return stored_cache[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            stored_cache[args] = result
+            return result
```

Comment:
U can use only one return here and return stored_cache[args]

Reply:
Made changes.

Also shortened:
`stored_cache[args] = func(*args)`

and changed annotation for wrapper:
`def wrapper(*args) -> dict:`

Accidentally deleted my last commit.

Reply:
Made changes.
Can you look up if it's ok?
Thank you for the review!